### PR TITLE
Re-enables some tests for linalg.det

### PR DESCRIPTION
**Pull request by lezcano**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #85141

At last.

Fixes https://github.com/pytorch/functorch/issues/961

### CODE REVIEW

File: functorch/test/test_ops.py [hunk 1]
```diff
@@ -825,10 +825,10 @@ def test_vmapjvpall(self, device, dtype, op):
         skip('to'),  # RuntimeError: required rank 4 tensor to use channels_last format
         xfail('nn.functional.huber_loss'),
         xfail('lu'),
-        skip('linalg.det', 'singular'),  # https://github.com/pytorch/functorch/issues/961
+        xfail('linalg.det'),
+        xfail('linalg.det', 'singular'),
```
Comment (zou3519) [inline functorch/test/test_ops.py:829]: Just to check, this doesn't completely fix https://github.com/pytorch/functorch/issues/961, right? It looks like previously we had vmapjvpall support for both of these, but not anymore: https://github.com/pytorch/pytorch/commit/67b104af02584ef60b8d26a0392947367767fbf1

At any rate we should still land this to get the skips out of the way.

**Review by zou3519 — APPROVED** [review]
awesome!

File: functorch/test/test_ops.py (hunk 1 above)
Comment (Lezcano) [reply functorch/test/test_ops.py:829]: It does not, as you have mentioned. Now, I don't think we need a separate issue to track it, as these are now xfails.

Comment (zou3519) [reply functorch/test/test_ops.py:829]: I'll put up a separate issue. This is technically a regression from our previous release

**Pull request closed without merging**